### DISCUSSION TITLE
Pimcore Academy, Value: Open Source

### DISCUSSION THREAD

**Discussion by NiklasBr** [pimcore/pimcore#16913] [General]
I have a question regarding this statement:

> Our software is based on open source. Which means that it is always accessible. So you can always check what we are doing. You can always check how our software works. And you will always be secured that you can get your data out of it.

But a lot of Pimcore's software are locked into hidden private repositories, using Pimcore Commercial License (PCL) *only* and have this copyright notice:

```php
/**
 * Pimcore
 *
 * This source file is available under following license:
 * - Pimcore Commercial License (PCL)
 *
 *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
 *  @license    http://www.pimcore.org/license     PCL
 */
```

That includes:
- Light Theme for Admin UI Classic
- Portal Engine
- Copilot
- Datahub File Export
- Simple REST API
- …

Given that, it seems not to be compatible with open source at all. Therefore the above quote seems incorrect to me. Can anyone explain how PCL is an [Open Source-compatible license](https://opensource.org/) and if I am incorrect?

**Comment by dpfaffenbauer** [comment]
I think the point is that the Pimcore Core is Open-Source.

**Reply by NiklasBr** [reply]
I didn't understand it as such, it was said in the PaaS context which is PCL only.

**Reply by dpfaffenbauer** [reply]
A bit of context for the statement would help ;)

**Reply by NiklasBr** [reply]
> A bit of context for the statement would help ;)

Agreed ;) 

It's in the first part of first chapter if you want to watch it, the background slide contains this and is clearly about the Platform:

![image](https://github.com/pimcore/pimcore/assets/279826/fe1bb39d-c82a-435f-8ad0-4992693d5a7d)

Unfortunately Pimcore has not provided subtitles so I cannot spend a lot of time transcribing the minutes before and after.

**Comment by drietsch** [comment]
Hello Niklas,

you've ventured into a realm where passion meets pragmatism, ideals brush against reality. The statement you're referring to—about the accessibility and openness of Pimcore—opens a Pandora's box of heated debates and fervent ideals within the tech community - just look at the Redis SSPL debates in the last few weeks, or at the desaster of Elastic.

Yes, at the heart of Pimcore lies a commitment to open-source—a beacon of transparency, collaboration, and innovation. This is not just a technicality; it's a principle. It's about the right to peer into the very soul of the software we use, to understand its inner workings, and to assert our freedom to modify and share it. The GNU General Public License (GPLv3) that underpins Pimcore's Community Edition isn't just a legal document; it's a manifesto—a declaration of the unyielding belief in the power of open-source.

But here's where the waters get choppy, and our ideals are tested. The Pimcore Commercial License (PCL) enters the scene—not as a villain in our story, but as a complex character whose motives are shaped by the harsh realities of sustaining a monumental project like Pimcore. This isn't about abandoning principles; it's about navigating the treacherous waters between open-source utopia and the economic realities that govern our world.

The notion of "open core" isn't a compromise; it's a bridge. It's a testament to the struggle of balancing ideals with sustainability. The core of Pimcore remains open, a testament to our belief in freedom and collaboration. Yet, the commercial extensions under PCL acknowledge the undeniable truth that passion doesn't pay the bills—sustainability does.

The debates around what constitutes "open source" are more than semantic squabbles; they're reflections of our deepest beliefs about technology and its role in society. Yes, the OSI has its definitions, but "open source" is a concept that transcends the confines of any single definition. It's a banner under which we rally for a more open, transparent, and collaborative world.

This conversation isn't just about licensing models; it's about the heart and soul of the software community. It's about finding a path that honors our ideals while acknowledging the realities of the world we live in. It's about the passion that drives us to build, share, and improve the software that shapes our world.

The journey of Pimcore, from its open-source roots to its embrace of a dual-licensing model, is a microcosm of the larger journey of the software industry. It's a journey fraught with challenges, but also filled with hope. Hope that we can find a way to sustain our ideals and our projects. Hope that we can continue to build a world that values openness, transparency, and collaboration.

Your voice, Niklas, and the voices of all who dwell in this realm of codes and licenses, are vital. They remind us that behind every line of code, every legal document, every commercial product, there are people—passionate, idealistic, pragmatic people—striving to make the world a better, more open place.

With passion and pragmatism,

Dietz

P.S.: Sorry, Niklas, for the longer post. This is truly important to me. So, to directly answer the question you originally referred to: I strongly believe it's a "Yes".

**Reply by NiklasBr** [reply]
> P.S.: Sorry, Niklas, for the longer post. This is truly important to me. So, to directly answer the question you originally referred to: I strongly believe it's a "Yes".

Can you give a motivation and specific example for how PCL is in any way Open Source?

**Reply by drietsch** [reply]
> Our software is based on open source. Which means that it is always accessible. So you can always check what we are doing. You can always check how our software works. And you will always be secured that you can get your data out of it.

I was referring to the quoted statement and explaining our dual license strategy and its reasoning. That quote holds true for all Pimcore Editions, which is why I affirmed with a "Yes" to fully agree with it. Regarding **the PCL, it is a proprietary software license** under which the source code is open and can be modified by customers without the obligation to release the modifications under the PCL, as opposed to GPLv3. It's not an OSI-accredited license but addresses the GPLv3 copyleft challenges for enterprise customers and supports our financial sustainability. The Pimcore Community Edition is fully open-source, based on GPLv3; meanwhile, the Pimcore Enterprise Edition, Pimcore Professional, and the Pimcore PaaS operate under the PCL.

Cheers, Dietz

**Reply by NiklasBr** [reply]
It seems to me you want to follow exactly the same thing as TimyMCE did, and Redis is doing, and Elastic Co did. It's a good business decision, but it also creates uncertainty for customer who hears that it is open source but then is met with a hidden and proprietary wall. If I was to preach the value of Pimcore's software as open source, I also need to know _exactly_ what this means, both in spirit and in letter.

Given that you say

> Pimcore Platform is only available under the PCL license

And you say:

> PCL is a proprietary license

How could I make this less ambiguous to a customer who wants to know what you mean when you say it is open source? They cannot see the source on GitHub. The license I send them indicates that it is not open source. Can I confidently say that they are allowed to download, keep, and modify, and create derivatives of that PCL software?

**Reply by drietsch** [reply]
Now, I've got your point. Thank you for the clarification; it's completely understandable. Apologies for any misunderstanding.

**Regarding your question about whether users are allowed to download, keep, modify, and create derivatives of PCL software, yes, you can confidently affirm this.**

The exception lies in OEM scenarios involving the redistribution of PCL-based products or competitive SaaS/PaaS offerings derived purely from PCL functionalities. The PCL license currently accompanies every Enterprise Edition offering, and we plan to include the official license document (PCL license) in the relevant GitHub repositories within the next few weeks. I will update this thread once that's completed. You might wonder why uploading a single text file takes several weeks. Indeed, it's a valid question. If you caught my Pimcore Licensing presentation at Pimcore Inspire two weeks ago, I announced that Pimcore Studio, the new UI for Pimcore, would also be released at no charge under an adapted version for the free use of Pimcore Studio. We are actively working on this, aiming to ensure everything is done correctly.

Your comparison of the PCL, established in 2013, with the TinyMCE case touches on two distinct matters. TinyMCE's shift from MIT to GPL represents a move from a permissive to a copyleft license, a model that the Pimcore Community Edition has operated under for over ten years. This change is different from the shifts seen with MongoDB, Redis, Elastic, & Co.

**Reply by drietsch** [reply]
Let me clarify directly, especially regarding your comment about me stating:

> Pimcore Platform is only available under the PCL license

This isn't accurate, and I apologize for any confusion. After looking through my comments in this thread, I can't seem to find where that was mentioned. It might just be my eyes playing tricks on me, especially after having eye laser surgery recently. Anyway, let's clear this up.

The Pimcore Platform is dual-licensed, and here's how:

- At its heart, the core of Pimcore, along with the entire Pimcore Community Edition, is licensed under the OSI-accredited GPLv3 license. This has been the steadfast approach for over a decade, and we're committed to keeping it this way. Simply put, Pimcore is free and open-source. Full stop.

- Beyond the Community Edition, we proudly offer additional functionalities and cloud services, such as Pimcore PaaS, under the commercially and enterprise-friendly PCL license. This license thoughtfully navigates the copyleft challenges presented by GPL, making things smoother for our enterprise users.
